aiidalab_widgets_base/viewers.py

```diff
@@ -580,9 +580,13 @@ def add_info(indx, atom):
 
         # Report dihedral angle and geometric center.
         if len(self.selection) == 4:
-            dihedral = self.structure.get_dihedral(self.selection) * 180 / np.pi
+            try:
+                dihedral = self.structure.get_dihedral(self.selection) * 180 / np.pi
+                dihedral_str = "{:.2f}".format(dihedral)
+            except ZeroDivisionError:
+                dihedral_str = "nan"
             return "{} atoms selected<br>Geometric center: ({})<br>Dihedral angle: {}".format(
-                len(self.selection), geom_center, dihedral.round(2))
+                len(self.selection), geom_center, dihedral_str)
```

Comment:
Thanks @eimrek. Just one little suggestion. We are now moving to f-strings, so please replace it with something like:
```python
return f"{len(self.selection)} atoms selected<br>Geometric center: ({geom_center})<br>Dihedral angle: {dihedral_str}"
```